File: app/src/main/java/com/jinhanexample/mvvmSample/java/author/data/AuthorEntity.java
package com.jinhanexample.mvvmSample.java.author.data;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "authorEntity")
public class AuthorEntity {

    @NonNull
    @PrimaryKey(autoGenerate = true)
    int id ;

    @ColumnInfo(name = "name")
    String name;

    @ColumnInfo(name = "books")
    String books;

    @ColumnInfo(name = "nation")
    String nation;

    public AuthorEntity(String name, String books, String nation) {
        this.name = name;
        this.books = books;
        this.nation = nation;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getBooks() {
        return books;
    }

    public void setBooks(String books) {
        this.books = books;
    }

    public String getNation() {
        return nation;
    }

    public void setNation(String nation) {
        this.nation = nation;
    }
}

File: app/src/main/java/com/jinhanexample/mvvmSample/kotlin/author/data/AuthorEntity.kt
package com.jinhanexample.mvvmSample.kotlin.author.data

import androidx.annotation.NonNull
import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

//데이터 클래스를 만들어주세요
//anotation을 이용해 roomDatabase를 만들 수 있습니다.
@Entity(tableName = "authorEntity") //테이블명 입력
data class AuthorEntity(

    //optional Int 타입으로 자동 키값 생성,
    @NonNull
    @PrimaryKey(autoGenerate = true)
    var id: Int?,

    //컬럼 구성
    @ColumnInfo(name = "name")
    var name: String,

    @ColumnInfo(name = "books")
    var books: String,

    @ColumnInfo(name = "nation")
    var nation: String

) {
    constructor() : this(null, "", "", "")
}